java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -265,8 +272,7 @@ private CanonicalOAuthAccessTokenResponseDto exchangeRefreshTokenForAccessToken(
 
             HttpResponse response = tokenRequest.execute();
             CanonicalOAuthAccessTokenResponseDto tokenResponse =
-                objectMapper.readerFor(CanonicalOAuthAccessTokenResponseDto.class)
-                    .readValue(response.getContent());
+                    tokenResponseParser.parseTokenResponse(response);
```

Comment:
Now with the interface, leaving in `TokenResponseParser` instance the responsibility of converting the response into `CanonicalOAuthAccessTokenResponseDto`